src/tx/applyTx/utils.test.js

```diff
@@ -154,10 +159,78 @@ describe('applyTx utils', () => {
         checkInsAndOuts(
           transfer,
           state,
+          {},
           ({ address }, i) => address === transfer.inputs[i].signer
         );
       }).toThrow(`Ins and outs values are mismatch for color ${colorERC721}`);
     });
+
+    test('With gasPrice', () => {
+      const colorERC20 = 0;
+      const depositERC20 = Tx.deposit(12, 500, ADDR_1, colorERC20);
+      const outpointERC20 = new Outpoint(depositERC20.hash(), 0);
+      const state = {
+        unspent: {
+          [outpointERC20.hex()]: depositERC20.outputs[0].toJSON(),
+        },
+      };
+      const transfer = Tx.transfer(
+        [new Input(outpointERC20)],
+        [new Output(300, ADDR_1, colorERC20)]
+      ).signAll(PRIV_1);
+      checkInsAndOuts(
+        transfer,
+        state,
+        { gasPrice: 0.02 },
+        ({ address }, i) => address === transfer.inputs[i].signer
+      );
+    });
+
+    test('Overpriced', () => {
+      const colorERC20 = 0;
+      const depositERC20 = Tx.deposit(12, 500, ADDR_1, colorERC20);
+      const outpointERC20 = new Outpoint(depositERC20.hash(), 0);
+      const state = {
+        unspent: {
+          [outpointERC20.hex()]: depositERC20.outputs[0].toJSON(),
+        },
+      };
+      const transfer = Tx.transfer(
+        [new Input(outpointERC20)],
+        [new Output(300, ADDR_1, colorERC20)]
+      ).signAll(PRIV_1);
+      expect(() => {
+        checkInsAndOuts(
+          transfer,
+          state,
+          { gasPrice: 0.01 },
```

Comment:
i think the gasPrice can only be whole numbers (we can not use decimals in solidity). I suggest to just increase the tx value, append 18*`0`, or something.